tests/views/test_admin.py

```diff
@@ -104,3 +130,8 @@ def test_climate_rec_edit(self):
         )
         form["associations"] = [admindiv.id for admindiv in admindivs]
         form.submit(status=302)
+
+    def test_preview(self):
+        resp = self.testapp.get("/preview/", status=302)
+        assert resp.location == "http://localhost/preview/en/"
```

Comment:
I guess tests run on default port 80?

Reply:
Tests only create a WSGI application not a webserver.
`testapp` create requests with appropriate environnement variables, send them to WSGI application which return responses.
But no, there is no port opened here.